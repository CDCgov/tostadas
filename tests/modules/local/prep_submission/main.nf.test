import XmlToJson

nextflow_process {

    name "Test Process PREP_SUBMISSION"
    script "modules/local/prep_submission/main.nf"
    process "PREP_SUBMISSION"

    test("Run with profile test, virus | biosample") {

        when {
            params {
                // define parameters here
                submission_prod_or_test = "test"
                send_submission_email = false
                biosample = true
                species = "virus"
                submission_mode = "ftp"
                output_dir = "${outputDir}"
                metadata_basename = "mpxv_test_metadata"
                overwrite_output = true
                dry_run = true
            }

            process {
                """
                // define inputs of the process here
                input[0] = [ [batch_id: 'batch_1', batch_tsv: file("${projectDir}/tests/modules/local/prep_submission/batch_1.tsv")], 
                             [ [meta: [ sample_id: 'NY0006' ], fq1: file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R1_001.fastq.gz"), fq2: file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R2_001.fastq.gz"), fasta: file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta"), nnp: null, gff: null], 
                               [meta: [ sample_id: 'IL0005' ], fq1: file("${projectDir}/assets/sample_fastqs/mpox/LIY15306A2_2022_054_3005007722.R_1.mpx.fastq.gz"), fq2: file("${projectDir}/assets/sample_fastqs/mpox/LIY15306A2_2022_054_3005007722.R_2.mpx.fastq.gz"), fasta: file("${projectDir}/assets/sample_fastas/mpox/IL0005.fasta"), nnp: null, gff: null]
                             ], 
                             ["biosample"] ]
                input[1] = file("${projectDir}/conf/submission_config.yaml")
                """
            }
        }

        then {
            // check if test case succeeded
            assert process.success
            // check that this run's submission.xml is the same as the last one (?: what I've passed is not a channel, the channel is a folder so can I use that or is it too complex)
            assert snapshot(XmlToJson.xmlToJson(file("${process.out.submission_dir[0]}/biosample/submission.xml"))).match()


        }
    }
}
