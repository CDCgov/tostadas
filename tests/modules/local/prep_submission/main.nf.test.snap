{
    "Run with profile test, virus | sra": {
        "content": [
            {
                "Description": {
                    "Comment": "Batch submission",
                    "Organization": {
                        "role": "owner",
                        "type": "institute",
                        "Name": "CDC",
                        "Contact": {
                            "email": "test@cdc.gov",
                            "Name": {
                                "First": "John",
                                "Last": "Doe"
                            }
                        }
                    }
                },
                "Action": [
                    {
                        "AddFiles": {
                            "target_db": "SRA",
                            "File": [
                                {
                                    "file_path": "NY0006_R1.fastq.gz",
                                    "DataType": "generic-data"
                                },
                                {
                                    "file_path": "NY0006_R2.fastq.gz",
                                    "DataType": "generic-data"
                                }
                            ],
                            "Attribute": [
                                {
                                    "name": "instrument_model"
                                },
                                {
                                    "name": "library_strategy"
                                },
                                {
                                    "name": "library_source"
                                },
                                {
                                    "name": "library_selection"
                                },
                                {
                                    "name": "library_layout"
                                },
                                {
                                    "name": "library_construction_protocol"
                                },
                                {
                                    "name": "library_name"
                                }
                            ],
                            "AttributeRefId": [
                                {
                                    "name": "BioProject",
                                    "RefId": {
                                        "PrimaryId": "PRJNA849962"
                                    }
                                },
                                {
                                    "name": "BioSample",
                                    "RefId": {
                                        "SPUID": {
                                            "spuid_namespace": ""
                                        }
                                    }
                                }
                            ],
                            "Identifier": {
                                "SPUID": {
                                    "spuid_namespace": ""
                                }
                            }
                        }
                    },
                    {
                        "AddFiles": {
                            "target_db": "SRA",
                            "File": [
                                {
                                    "file_path": "IL0005_R1.fastq.gz",
                                    "DataType": "generic-data"
                                },
                                {
                                    "file_path": "IL0005_R2.fastq.gz",
                                    "DataType": "generic-data"
                                }
                            ],
                            "Attribute": [
                                {
                                    "name": "instrument_model"
                                },
                                {
                                    "name": "library_strategy"
                                },
                                {
                                    "name": "library_source"
                                },
                                {
                                    "name": "library_selection"
                                },
                                {
                                    "name": "library_layout"
                                },
                                {
                                    "name": "library_construction_protocol"
                                },
                                {
                                    "name": "library_name"
                                }
                            ],
                            "AttributeRefId": [
                                {
                                    "name": "BioProject",
                                    "RefId": {
                                        "PrimaryId": "PRJNA849962"
                                    }
                                },
                                {
                                    "name": "BioSample",
                                    "RefId": {
                                        "SPUID": {
                                            "spuid_namespace": ""
                                        }
                                    }
                                }
                            ],
                            "Identifier": {
                                "SPUID": {
                                    "spuid_namespace": ""
                                }
                            }
                        }
                    }
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:22:17.141335514"
    },
    "Run with profile test, virus | biosample": {
        "content": [
            {
                "Description": {
                    "Comment": "Batch submission",
                    "Organization": {
                        "role": "owner",
                        "type": "institute",
                        "Name": "CDC",
                        "Contact": {
                            "email": "test@cdc.gov",
                            "Name": {
                                "First": "John",
                                "Last": "Doe"
                            }
                        }
                    }
                },
                "Action": [
                    {
                        "AddData": {
                            "target_db": "BioSample",
                            "Data": {
                                "content_type": "xml",
                                "XmlContent": {
                                    "BioSample": {
                                        "schema_version": "2.0",
                                        "SampleId": {
                                            "SPUID": {
                                                "spuid_namespace": ""
                                            }
                                        },
                                        "Descriptor": {
                                            "Title": "Test Monkeypoxvirus"
                                        },
                                        "Organism": {
                                            "OrganismName": "Monkeypox Virus"
                                        },
                                        "BioProject": {
                                            "PrimaryId": {
                                                "db": "BioProject"
                                            }
                                        },
                                        "Package": "SARS-CoV-2.wwsurv.1.0",
                                        "Attributes": {
                                            "Attribute": [
                                                {
                                                    "attribute_name": "isolate"
                                                },
                                                {
                                                    "attribute_name": "host_disease"
                                                },
                                                {
                                                    "attribute_name": "host"
                                                },
                                                {
                                                    "attribute_name": "collected_by"
                                                },
                                                {
                                                    "attribute_name": "lat_lon"
                                                },
                                                {
                                                    "attribute_name": "geo_loc_name"
                                                },
                                                {
                                                    "attribute_name": "sample_type"
                                                },
                                                {
                                                    "attribute_name": "collection_date"
                                                },
                                                {
                                                    "attribute_name": "isolation_source"
                                                },
                                                {
                                                    "attribute_name": "host_age"
                                                },
                                                {
                                                    "attribute_name": "host_sex"
                                                },
                                                {
                                                    "attribute_name": "race"
                                                },
                                                {
                                                    "attribute_name": "ethnicity"
                                                }
                                            ]
                                        }
                                    }
                                }
                            },
                            "Identifier": {
                                "SPUID": {
                                    "spuid_namespace": ""
                                }
                            }
                        }
                    },
                    {
                        "AddData": {
                            "target_db": "BioSample",
                            "Data": {
                                "content_type": "xml",
                                "XmlContent": {
                                    "BioSample": {
                                        "schema_version": "2.0",
                                        "SampleId": {
                                            "SPUID": {
                                                "spuid_namespace": ""
                                            }
                                        },
                                        "Descriptor": {
                                            "Title": "Test Monkeypoxvirus"
                                        },
                                        "Organism": {
                                            "OrganismName": "Monkeypox Virus"
                                        },
                                        "BioProject": {
                                            "PrimaryId": {
                                                "db": "BioProject"
                                            }
                                        },
                                        "Package": "SARS-CoV-2.wwsurv.1.0",
                                        "Attributes": {
                                            "Attribute": [
                                                {
                                                    "attribute_name": "isolate"
                                                },
                                                {
                                                    "attribute_name": "host_disease"
                                                },
                                                {
                                                    "attribute_name": "host"
                                                },
                                                {
                                                    "attribute_name": "collected_by"
                                                },
                                                {
                                                    "attribute_name": "lat_lon"
                                                },
                                                {
                                                    "attribute_name": "geo_loc_name"
                                                },
                                                {
                                                    "attribute_name": "sample_type"
                                                },
                                                {
                                                    "attribute_name": "collection_date"
                                                },
                                                {
                                                    "attribute_name": "isolation_source"
                                                },
                                                {
                                                    "attribute_name": "host_age"
                                                },
                                                {
                                                    "attribute_name": "host_sex"
                                                },
                                                {
                                                    "attribute_name": "race"
                                                },
                                                {
                                                    "attribute_name": "ethnicity"
                                                }
                                            ]
                                        }
                                    }
                                }
                            },
                            "Identifier": {
                                "SPUID": {
                                    "spuid_namespace": ""
                                }
                            }
                        }
                    }
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-19T19:05:30.223127829"
    }
}